acme/acme/crypto_util.py

```diff
@@ -366,6 +367,78 @@ def _pyopenssl_extract_san_list_raw(cert_or_req: Union[crypto.X509, crypto.X509R
     return sans_parts
 
 
+def make_self_signed_cert(key: crypto.PKey, domains: Optional[List[str]] = None,
+                          not_before: Optional[int] = None,
+                          validity: int = (7 * 24 * 60 * 60), force_san: bool = True,
+                          extensions: Optional[List[x509.Extension]] = None,
+                          ips: Optional[List[Union[ipaddress.IPv4Address,
+                                                   ipaddress.IPv6Address]]] = None
+                          ) -> crypto.X509:
```

Comment:
Similar to my comment above...while we're migrating, could it make sense to return a `cryptography.X509.certificate` instead in our new method? Or would that be very annoying to thread through things?